instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/http/HttpClientAttributesExtractor.java

```diff
@@ -145,4 +163,57 @@ private static String stripSensitiveData(@Nullable String url) {
     }
     return url.substring(0, schemeEndIndex + 3) + "REDACTED:REDACTED" + url.substring(atIndex);
   }
+
+  private static String redactQueryParameters(String url) {
+
+    int questionMarkIndex = url.indexOf('?');
+    if (questionMarkIndex == -1 || !containsParamToRedact(url)) {
+      return url;
+    }
+
+    StringBuilder redactedParameters = new StringBuilder();
+    boolean paramToRedact = false; // To be able to skip the characters of the parameters to redact
+    boolean paramNameDetected = false;
+    boolean reference = false;
+
+    // To build a parameter name until we reach the '=' character
+    // If the parameter name is a one to redact, we will redact the value
+    StringBuilder currentParamName = new StringBuilder();
+
+    for (int i = questionMarkIndex + 1; i < url.length(); i++) {
+      char currentChar = url.charAt(i);
+      if (currentChar == '=') {
+        paramNameDetected = true;
+        redactedParameters.append(currentParamName);
+        redactedParameters.append('=');
+        if (PARAMS_TO_REDACT.contains(currentParamName.toString())) {
+          redactedParameters.append("REDACTED");
+          paramToRedact = true;
+        }
+      } else if (currentChar == '&') { // New parameter delimiter
+        redactedParameters.append('&');
+        paramNameDetected = false;
+        paramToRedact = false;
+        currentParamName.setLength(
+            0); // To avoid creating a new StringBuilder for each new parameter
+      } else if (currentChar == '#') { // Reference delimiter
+        reference = true;
+        redactedParameters.append('#');
```

Comment:
`break` here instead? (no need to continue looking for param key/values after the `#`)

Reply:
this is what I meant: https://github.com/jeanbisutti/opentelemetry-java-instrumentation/pull/12

Reply:
It's what I have understood but I had this concern https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/13114#discussion_r1954818526

If have merged your proposal because you seem to prefer this one.

Reply:
where is regex used in https://github.com/jeanbisutti/opentelemetry-java-instrumentation/pull/12?